apps/ledger-live-mobile/src/newArch/features/WalletSync/hooks/useSyncWithQrCode.ts

```diff
@@ -0,0 +1,74 @@
+import { useCallback, useState } from "react";
+import { MemberCredentials } from "@ledgerhq/trustchain/types";
+import { createQRCodeCandidateInstance } from "@ledgerhq/trustchain/qrcode/index";
+import { InvalidDigitsError } from "@ledgerhq/trustchain/errors";
+import { setTrustchain } from "@ledgerhq/trustchain/store";
+import { useDispatch } from "react-redux";
+import { useNavigation } from "@react-navigation/native";
+import { Platform } from "react-native";
+import { Steps } from "../types/Activation";
+import { NavigatorName, ScreenName } from "~/const";
+
+export const useSyncWithQrCode = () => {
+  const [digits, setDigits] = useState<number | null>(null);
+  const [input, setInput] = useState<string | null>(null);
+
+  const navigation = useNavigation();
+
+  const [inputCallback, setInputCallback] = useState<((input: string) => void) | null>(null);
+  const dispatch = useDispatch();
+
+  const onRequestQRCodeInput = useCallback(
+    (config: { digits: number }, callback: (input: string) => void) => {
+      setDigits(config.digits);
+      setInputCallback(() => callback);
+    },
+    [],
+  );
+
+  const onSyncFinished = useCallback(() => {
+    setDigits(null);
+    setInput(null);
+    setInputCallback(null);
+    navigation.navigate(NavigatorName.WalletSync, {
+      screen: ScreenName.WalletSyncSuccess,
+      params: {
+        created: false,
+      },
+    });
+  }, [navigation]);
+
+  const handleStart = useCallback(
+    async (
+      url: string,
+      memberCredentials: MemberCredentials,
+      setCurrentStep: (step: Steps) => void,
+    ) => {
+      try {
+        const trustchain = await createQRCodeCandidateInstance({
+          memberCredentials,
+          scannedUrl: url,
+          memberName: Platform.OS + " " + Platform.Version,
```

Comment:
in `hooks/useTrustchainSdk` we do

```
    const name = `${platformMap[Platform.OS] ?? Platform.OS} ${Platform.Version} ${hash ? " " + hash : ""}`;
```

can we factorize and use the same code? thanks

Reply:
Indeed Nice catch

Reply:
Done ✅